exporter/views.py

```diff
@@ -8,57 +7,48 @@
 from django.http import HttpResponse
 from django.utils.decorators import method_decorator
 from django.views.generic import TemplateView
+from tabular_export.core import export_to_csv_response, flatten_queryset
 
-from concordia.models import Asset, Campaign, Transcription, UserAssetTagCollection
+from concordia.models import Asset, Campaign, Transcription
 from concordia.storage import ASSET_STORAGE
 
 
 class ExportCampaignToCSV(TemplateView):
     """
-    Exports the transcription and tags to csv file
+    Exports the most recent transcription for each asset in a campaign
 
     """
 
-    template_name = "transcriptions/campaign.html"
-
     @method_decorator(login_required)
     def get(self, request, *args, **kwargs):
-        campaign = Campaign.objects.get(slug=self.kwargs["campaign_slug"])
-        asset_list = Asset.objects.filter(item__project__campaign=campaign).order_by(
-            "title", "sequence"
-        )
-        # Create the HttpResponse object with the appropriate CSV header.
-        response = HttpResponse(content_type="text/csv")
-        response["Content-Disposition"] = 'attachment; filename="{0}.csv"'.format(
-            campaign.slug
+        qs = Campaign.objects.filter(slug=self.kwargs["campaign_slug"]).select_related()
+        headers, data = flatten_queryset(
+            qs,
+            field_names=[
+                "title",
+                "project__title",
+                "project__item__title",
+                "project__item__item_id",
+                "project__item__asset__title",
+                "project__item__asset__transcription_status",
+                "project__item__asset__download_url",
+                "project__item__asset__transcription__text",
```

Comment:
According to https://support.office.com/en-us/article/excel-specifications-and-limits-1672b34d-7043-467e-8e27-269d656771c3 Excel can't handle more than 32K characters in a table cell. Currently the largest transcription appears to have 17K characters so it's not inconceivable that we'd approach that limit. Do we care?

Reply:
I think so. I believe we should detect when a transcription has to be truncated to fit into a CSV and report a warning or error if so.

Reply:
It would only be an issue when using Excel rather than other tools so perhaps we could just do a single max length query and if it's near 32K call messages.warning with a note to use caution with Excel?